src/tasks/tasks.hpp

```diff
@@ -270,18 +283,23 @@ class TaskList {
   template <class Arg1, class... Args>
   TaskID AddTask(const TaskQualifier tq, TaskID dep, Arg1 &&arg1, Args &&...args) {
     if constexpr (std::is_invocable<Arg1, Args...>::value) {
-      return AddTaskImpl(tq, dep, {}, std::forward<Arg1>(arg1),
+      return AddTaskImpl(tq, dep, {}, 0, std::forward<Arg1>(arg1),
                          std::forward<Args>(args)...);
     } else if constexpr (std::is_convertible<Arg1, std::string>::value) {
-      return AddTaskImpl(tq, dep, std::forward<Arg1>(arg1), std::forward<Args>(args)...);
+      return AddTaskImpl(tq, dep, std::forward<Arg1>(arg1), 0,
+                         std::forward<Args>(args)...);
+    } else if constexpr (is_tuple_t<Arg1>::value) {
+      return AddTaskImpl(tq, dep, std::get<0>(arg1), std::get<1>(arg1), std::get<2>(arg1),
+                         std::forward<Args>(args)...);
```

Comment:
Allow passing a `tuple` containing the name of the task, the verbosity level of the task, and the task function itself. Allows easy print out of the order in which tasks are run.